containers/Ideas/IdeaCreate.js

```diff
@@ -22,34 +22,108 @@ import API from '../../BaseApi';
 
 export const IdeaCreate = props => {
 
-  const [error, setError] = useState();
-  const {moduleId, project} = props.route.params;
+  const {idea, project, editing} = props.route.params;
+  const moduleId = project.single_agenda_setting_module;
   const arrowLeftIcon = (
     <IconSLI name='arrow-left' size={22} />
   );
+  const ideaValidationSchema = yup.object().shape({
+    name: yup
+      .string()
+      .required('Idea title is Required'),
+    description: yup
+      .string()
+      .min(20, ({ min }) => `Description must be at least ${min} characters`)
+      .max(100, 'Description must be no longer then 100 characters')
+      .required('Description is required'),
+  });
+
+  const [error, setError] = useState();
+  const [clicked, setClicked] = useState(false);
+  const [categories, setCategories] = useState([]);
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [labels, setLabels] = useState([]);
+  const [labelsChecked, setLabelsChecked] = useState([]);
 
   useEffect(() => {
     if (error) {
       Alert.alert('An error occured', error, [{ text: 'Ok' }]);
     }
     else {
-      fetchLabelsAndCategories();
+      setLabelsAndCategories();
+    }
+  }, [error, project]);
+
+  const setLabelsAndCategories = async() => {
+    const module = await API.getModule(moduleId);
+
+    if (module.categories) {
+      // map property names to format needed for formik
+      setCategories(module.categories.map(category => ({value: category.id, label: category.name})));
+      if (editing) {
+        setSelectedCategory(idea.category.id);
+      }
+      else {
+        setSelectedCategory(module.categories[0].id);
+      }
     }
-  }, [error, categories, labels]);
+    if (module.labels) {
+      setLabels(module.labels.map(label => ({value: label.id, label: label.name})));
+
+      if (editing) {
+        // idea.labels is array of label ids -> map that to array of bools with length = number of
+        // of module labels and true iff label is checked
+        setLabelsChecked(module.labels.map(label => (idea.labels.some(entry => (entry.id===label.id) ? true : false))));
```

Comment:
@kha Not sure if I understand what you mean, but I would be happy to facilitate this statement!

In case you mean instead of 
`idea.labels.some(entry => (entry.id===label.id) ? true : false))`  just  
`idea.labels.map(entry => (entry.id===label.id) ? true : false))`

that would not work I think, because the first one returns just one value true/false if label is contained in idea.labels and the second one returns a list.

But probably you meant sth else?

Reply:
i thought the following would result in the same, but not sure here:
So instead of:
`module.labels.map(label => (idea.labels.some(entry => (entry.id===label.id) ? true : false)))`
that one:
`module.labels.map(label => (idea.labels.some(entry => entry.id===label.id)))`

Reply:
It totally does, yes! Thank you!! I guess I have been staring too much at this :sweat_smile: 
Its changed now..